src/Backend/Component/Core/ConfigurationStep/Website.php

```diff
@@ -359,8 +426,28 @@ protected function createLayouts(int $themeId, array $modules): array
             ['mod' => $modules['wem_sg_footer']->id, 'col' => 'footer', 'enable' => '1'],
         ];
         $script = file_get_contents(Util::getPublicOrWebDirectory().'/bundles/wemsmartgear/examples/code_javascript_personnalise_1.js');
-        $script = str_replace('{{config.googleFonts}}', "'".implode("','", $config->getSgGoogleFonts())."'", $script);
+        if (\count($config->getSgGoogleFonts()) > 0) {
+            $script = str_replace('{{config.googleFonts}}', "'".implode("','", $config->getSgGoogleFonts())."'", $script);
+        } else {
+            $script = preg_replace('/\/\/ -- GFONT(.*)\/\/ -- \/GFONT/s', '', $script);
+        }
+
         $script = str_replace('{{config.framway.path}}', $config->getSgFramwayPath(), $script);
+        switch ($config->getSgAnalytics()) {
```

Comment:
@LupusVII here is the code removing the unecessary pieces of code for analytics solutions (same logic applied for Google Fonts if none are used)